File: Cplusplus/0078_subsets/0078_subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> res;
        vector<int> list;
        if (nums.size() == 0) {
            res.push_back(list);
            return res;
        }
        sort(nums.begin(), nums.end());
        backtracking(res, list, nums, 0);
        return res;
    }
    
    void backtracking(vector<vector<int>> &res, vector<int> list, vector<int> &nums, int pos) {
        res.push_back(list);    
        for (int i = pos; i < nums.size(); i++) {
            list.push_back(nums[i]);
            backtracking(res, list, nums, i + 1);
            list.pop_back();
        }
    }
};

File: Python/0078_subsets/0078_subsets.py
class Solution:
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        res = []
        self.backtracking([], nums, res)
        return res
               
    def backtracking(self, solution , nums, res):
        res.append(solution)
        if not nums:
            return
        for i in range(len(nums)):
            temp = solution + [nums[i]]
            self.backtracking(temp, nums[i + 1:], res)
